lib/platform/github/massage-markdown-links.ts

```diff
@@ -0,0 +1,81 @@
+import is from '@sindresorhus/is';
+import remark from 'remark';
+import type { Plugin } from 'unified';
+import { logger } from '../../logger';
+
+interface UrlMatch {
+  start: number;
+  end: number;
+  replaceTo: string;
+}
+
+const urlRegex =
+  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-_a-z0-9]+\/[-_a-z0-9]+\/(?:discussions|issues|pull)\/[0-9]+(?:#[-_a-z0-9]+)?/i;
+
+function massageLink(input: string): string {
+  return input.replace(/(?:to)?github\.com/i, 'togithub.com');
+}
+
+function collectLinkPosition(input: string, matches: UrlMatch[]): Plugin<any> {
+  const transformer = (tree: any): void => {
```

Comment:
Dont' we have types for tree? 😕

Reply:
Seems like it's a pretty big mismatch between types and real objects I saw in debugger

Reply:
Can we add our own minimal tree interface then?

Reply:
```suggestion
  const transformer = (tree: TreeNode): void => {
```

Can this work?